src/components/NetworkSelector.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+
+interface NetworkSelectorProps {
+    network: string;
+    setNetwork: (network: string) => void;
+}
+
+export class NetworkSelector extends React.Component<NetworkSelectorProps, Object> {
+
+    constructor(props: NetworkSelectorProps) {
+        super(props);
+        this.state = {};
+        this.handleNetworkUpdate = this.handleNetworkUpdate.bind(this);
```

Comment:
Coming from other languages, I just have a very hard time defining a member function in the constructor. Since the component is encapsulating a member function, I'm refraining from building a functional component. Also, sadly, TypeScript complains loudly if I change the type of the state to 'void'.